blocks/plp/plp.js

```diff
@@ -0,0 +1,528 @@
+ * @returns {HTMLElement} Product card element
+ */
+function createProductCard(product, ph) {
+  const card = document.createElement('div');
+  card.className = 'plp-product-card';
+
+  const image = createProductImage(product);
+  const title = createProductTitle(product);
+  const price = createProductPrice(product);
+  const colors = createProductColors(product);
+  const viewDetails = createProductButton(product, ph, 'View Details', 'emphasis');
+  const compare = createProductButton(product, ph, 'Compare');
+
+  card.append(image, title, price, colors, viewDetails, compare);
+  card.addEventListener('click', () => {
+    window.location.href = product.path;
+  });
+
+  return card;
+}
+
+/**
+ * Styles product content for display in a carousel view.
+ * @param {Object} content - Product content
+ * @param {Object} ph - Placeholder object with localized text strings
+ */
+async function styleRowAsSlide(content, ph) {
+  const [image, body] = content.children;
+  const link = body.querySelector('a[href]');
+  const { pathname } = new URL(link.href);
+  const [product] = await lookupProducts([pathname]);
+
+  // product image
+  let img = image.querySelector('picture');
+  if (!img) {
+    img = createProductImage(product);
+    image.replaceChildren(img);
+  }
+
+  // product title as a link to PDP
+  let title = body.querySelector('h3');
+  if (!title) {
+    title = createProductTitle(product, 'h3');
+    body.prepend(title);
+  }
+
+  // authored content
+  const ps = body.querySelectorAll('p');
+  ps.forEach((p) => {
+    const a = p.querySelector('a[href]');
+    if (a) p.remove();
+    else {
+      const [brow, ...text] = p.textContent.split(':').map((t) => t.trim());
+      const eyebrow = document.createElement('p');
+      eyebrow.className = 'eyebrow';
+      eyebrow.textContent = brow;
+      p.textContent = text;
+      p.parentNode.insertBefore(eyebrow, p);
+    }
+  });
+
+  // color options
+  const colors = createProductColors(product);
+  const colorOptions = document.createElement('p');
+  colorOptions.className = 'eyebrow';
+  colorOptions.textContent = ph.colorOptions || 'Color options';
+  body.append(colorOptions, colors);
+
+  // starting at price
+  if (product.price) {
+    const startingAt = document.createElement('p');
+    startingAt.className = 'eyebrow';
+    startingAt.textContent = ph.startingAt || 'Starting at';
+
+    const price = createProductPrice(product);
+    body.append(startingAt, price);
+  }
+
+  // "Shop Now" button
+  const shopNow = createProductButton(product, ph, 'Shop Now');
+  body.appendChild(shopNow);
+}
+
+/**
+ * Builds a product carousel from a block containing product links.
+ * @param {HTMLElement} block - Block element
+ * @param {Object} ph - Placeholder object with localized text strings
+ * @returns {Promise<void>}
+ */
+async function buildProductCarousel(block, ph) {
+  const rows = [...block.children];
+  rows.forEach(async (row) => {
+    await styleRowAsSlide(row, ph);
```

Comment:
`forEach` won't respect the `await`. Probably not a problem but would be good to fix.